pyo3-macros-backend/src/pyclass.rs

```diff
@@ -2031,6 +2031,8 @@ impl<'a> PyClassImplsBuilder<'a> {
             impl #pyo3_path::PyClass for #cls {
                 type Frozen = #frozen;
             }
+
+            impl crate::sealed::Sealed for crate::impl_::pyclass::PyClassImplCollector<#cls> {}
```

Comment:
This is not possible. The point of the sealing is that a trait is not implementable outside of PyO3. If we need to implement it for external types as part of the macro code, it can't be sealed. In that case the documentation/being part of `impl_` has to suffice making clear that it's an implementation detail and is not part of semver.